src/basic/terminal-util.c

```diff
@@ -1279,6 +1279,37 @@ int vt_reset_keyboard(int fd) {
         return 0;
 }
 
+int vt_restore(int fd) {
+        static const struct vt_mode mode = {
+                .mode = VT_AUTO,
+        };
+        int r, q = 0;
+
+        r = ioctl(fd, KDSETMODE, KD_TEXT);
+        if (r < 0)
+                q = log_debug_errno(errno, "Failed to set VT in text mode, ignoring: %m");
+
+        r = vt_reset_keyboard(fd);
+        if (r < 0) {
+                log_debug_errno(r, "Failed to reset keyboard mode, ignoring: %m");
+                q = q ?: r;
+        }
+
+        r = ioctl(fd, VT_SETMODE, &mode);
+        if (r < 0) {
+                log_debug_errno(r, "Failed to set VT_AUTO mode, ignoring: %m");
+                q = q ?: r;
+        }
+
+        r = fchown(fd, 0, (gid_t) -1);
+        if (r < 0) {
+                log_debug_errno(-errno, "Failed to chown VT, ignoring: %m");
```

Comment:
no need to negate errno for the first arg. log_debug_erro() does that on its own if necessary